contracts/kleros/TalentLayerMultipleArbitrableTransaction.sol

```diff
@@ -2,109 +2,276 @@
+     * @dev Id 0 is reserved to the protocol balance & address(0) to ETH balance
+     */
+    mapping(uint256 => mapping(address => uint256)) private platformIdToTokenToBalance;
+
+    /**
+     * @notice Instance of ServiceRegistry.sol
+     */
+    IServiceRegistry private serviceRegistryContract;
+
+    /**
+     * @notice Instance of TalentLayerID.sol
+     */
+    ITalentLayerID private talentLayerIdContract;
+
+    /**
+     * @notice Instance of TalentLayerPlatformID.sol
+     */
+    ITalentLayerPlatformID private talentLayerPlatformIdContract;
+
+    /**
+     * @notice Percentage paid to the protocol (per 10,000, upgradable)
+     */
+    uint16 public protocolFeePerTenThousand;
+
+    /**
+     * @notice Percentage paid to the platform who onboarded the user (per 10,000, upgradable)
+     */
+    uint16 public originPlatformFeePerTenThousand;
+
+    /**
+     * @notice Wallet which will receive the protocol fees
+     */
+    address payable private protocolWallet;
 
     // =========================== Constructor ==============================
 
-    /** @dev Called on contract deployment
-     *  @param _serviceRegistryAddress Contract address to ServiceRegistry.sol
-     *  @param _talentLayerIDAddress Contract address to TalentLayerID.sol
-     *  @param _arbitrator The arbitrator of the contract.
-     *  @param _arbitratorExtraData Extra data for the arbitrator.
-     *  @param _feeTimeout Arbitration fee timeout for the parties.
+    /**
+     * @dev Called on contract deployment
+     * @param _serviceRegistryAddress Contract address to ServiceRegistry.sol
+     * @param _talentLayerIDAddress Contract address to TalentLayerID.sol
+     * @param _talentLayerPlatformIDAddress Contract address to TalentLayerPlatformID.sol
+     * @param _arbitrator The arbitrator of the contract.
+     * @param _arbitratorExtraData Extra data for the arbitrator.
+     * @param _feeTimeout Arbitration fee timeout for the parties.
      */
     constructor(
         address _serviceRegistryAddress,
         address _talentLayerIDAddress,
-        Arbitrator _arbitrator, 
+        address _talentLayerPlatformIDAddress,
+        Arbitrator _arbitrator,
         bytes memory _arbitratorExtraData,
         uint _feeTimeout
     ) {
-        _setServiceRegistryAddress(_serviceRegistryAddress);
-        _setTalentLayerIDAddress(_talentLayerIDAddress);
+        serviceRegistryContract = IServiceRegistry(_serviceRegistryAddress);
+        talentLayerIdContract = ITalentLayerID(_talentLayerIDAddress);
+        talentLayerPlatformIdContract = ITalentLayerPlatformID(_talentLayerPlatformIDAddress);
+        protocolFeePerTenThousand = 100;
+        originPlatformFeePerTenThousand = 200;
+        protocolWallet = payable(owner());
         // arbitrator = _arbitrator;
         // arbitratorExtraData = _arbitratorExtraData;
         // feeTimeout = _feeTimeout;
+
+        emit ProtocolFeeUpdated(protocolFeePerTenThousand);
```

Comment:
I thought we could do the same as for Token Decimals and store these values in The Graph, to avoid making an RPC call for each fee calculation. That's why I called these events even for the first init.

What do you think ?

Reply:
Ok cool. I'll leave this here for now then.